db/migrate/20220202140507_reorder_project_children.rb

```diff
@@ -1,33 +0,0 @@
-#-- copyright
-# OpenProject is an open source project management software.
-# Copyright (C) 2012-2024 the OpenProject GmbH
-#
-# This program is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License version 3.
-#
-# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
-# Copyright (C) 2006-2013 Jean-Philippe Lang
-# Copyright (C) 2010-2013 the ChiliProject Team
-#
-# This program is free software; you can redistribute it and/or
-# modify it under the terms of the GNU General Public License
-# as published by the Free Software Foundation; either version 2
-# of the License, or (at your option) any later version.
-#
-# This program is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with this program; if not, write to the Free Software
-# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
-#
-# See COPYRIGHT and LICENSE files for more details.
-#++
-
-class ReorderProjectChildren < ActiveRecord::Migration[6.1]
-  def up
-    ::Projects::ReorderHierarchyJob.perform_later
```

Comment:
Why did you remove this migration? This would still be necessary on old installations. I can see the point about removing delayed job cron removals, but this should remain

Reply:
The migration now fails as good job is not yet defined. You can just run the job in the migration without going through good_job

Reply:
Good point!
I wonder if that is okay to have it asynchronous here. Will the job be executed eventually if this migration happens alongside with GoodJob migration? 
I expect the code to be new when running this migration. It means that `::Projects::ReorderHierarchyJob.perform_later` will try to reach GoodJob tables which are absent at the moment, because good_job migrations are younger. If that is true there will be an error. I suggest to run the job synchronously to avoid the error then. Does it make sense to you?

By removing I tried to solve this problem which apparently is not a solution :)

Reply:
Let's rewrite it like that? I am not aware of how heavy the job can be...
```
class ReorderProjectChildren < ActiveRecord::Migration[6.1]
  def up
    ::Projects::ReorderHierarchyJob.new.perform
  end
end  
```